compiler+runtime/src/cpp/jank/runtime/obj/integer_range.cpp

```diff
@@ -0,0 +1,193 @@
+  {
+    return val <= end;
+  }
+
+  obj::integer_range::static_object(native_integer const end)
+    : end{ end }
+    , step{ 1 }
+    , bounds_check{ positive_step_bounds_check }
+  {
+  }
+
+  obj::integer_range::static_object(native_integer const start, native_integer const end)
+    : start{ start }
+    , end{ end }
+    , step{ 1 }
+    , bounds_check{ positive_step_bounds_check }
+  {
+  }
+
+  obj::integer_range::static_object(native_integer const start,
+                                    native_integer const end,
+                                    native_integer const step)
+    : start{ start }
+    , end{ end }
+    , step{ step }
+    , bounds_check{ step > 0 ? positive_step_bounds_check : negative_step_bounds_check }
+  {
+  }
+
+  obj::integer_range::static_object(native_integer const start,
+                                    native_integer const end,
+                                    native_integer const step,
+                                    obj::integer_range::bounds_check_t const bounds_check)
+    : start{ start }
+    , end{ end }
+    , step{ step }
+    , bounds_check{ bounds_check }
+  {
+  }
+
+  object_ptr obj::integer_range::create(native_integer const end)
+  {
+    if(end)
+    {
+      return make_box<obj::integer_range>(0, end, 1, positive_step_bounds_check);
+    }
+    return obj::persistent_list::empty();
+  }
+
+  object_ptr obj::integer_range::create(native_integer const start, native_integer const end)
+  {
+    return create(start, end, 1);
+  }
+
+  object_ptr obj::integer_range::create(native_integer const start,
+                                        native_integer const end,
+                                        native_integer const step)
+  {
+    if((step > 0 && end < start) || (step < 0 && start < end) || start == end)
+    {
+      return obj::persistent_list::empty();
+    }
+    /* TODO: Repeat object. */
+    //else if(is_zero(step))
+    //{ return make_box<obj::repeat>(start); }
+    return make_box<obj::integer_range>(start,
+                                        end,
+                                        step,
+                                        step > 0 ? positive_step_bounds_check
+                                                 : negative_step_bounds_check);
+  }
+
+  obj::integer_range_ptr obj::integer_range::seq()
+  {
+    return this;
+  }
+
+  obj::integer_range_ptr obj::integer_range::fresh_seq() const
+  {
+    return make_box<obj::integer_range>(start, end, step, bounds_check);
+  }
+
+  object_ptr obj::integer_range::first() const
+  {
+    return make_box<obj::integer>(start);
+  }
+
+  obj::integer_range_ptr obj::integer_range::next() const
+  {
+    if(count() < 1)
+    {
+      return nullptr;
+    }
+    return make_box<obj::integer_range>(start + step, end, step, bounds_check);
+  }
+
+  obj::integer_range_ptr obj::integer_range::next_in_place()
+  {
+    if(count() <= 1)
+    {
```

Comment:
In `next`, you check `count() < 1`, but in `next_in_place`, you check `count() <= 1`. Which is it you want? These fns should behave the same, excepting the mutation.

Reply:
Similar reasoning to the comment above.